### DISCUSSION TITLE
How to declare static data members of types?

### DISCUSSION THREAD

**Discussion by jamadagni** [hsutter/cppfront#1248] [Q&A]
Hello. I know I can declare a static function member of a type just by omitting the `this` parameter. But how to declare a static data member, such as a list of month names for a `Month` type, without having it being duplicated for all instances of the type?

```
Month : type = {
    id_ : int;
    names : std::vector<std::string> = ("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec");
}
```

**Comment by MaxSagebaum** [comment]
This is currently not possible.

**Comment by zaucy** [comment]
If you make `names` a constexpr it seems to work (tested on latest msvc and latest cppfront)

```cpp2
Month: type = {
	id_: int;
	names: std::vector<std::string> == ("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec");
}

main: () = {
	std::println("hello {}", Month::names.at(0));
}
```

**Reply by jamadagni** [reply]
Nice idea, but this will work only for static readonly data like in the above example.

How about static writable data? Those are useful too no?

**Reply by hsutter** [reply]
Static mutable data is a source of concurrency problems, and I've tried to minimize the ability to write that.

I do allow mutable global variables, but I want to ban them and require using Meyers singletons instead, but I haven't decided on whether I want magic statics to be the syntax for that yet.
